src/api/lexer/lexing.rs

```diff
@@ -0,0 +1,83 @@
+// This file is part of rust-web/twig
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use api::Position;
+use api::tokens::TokenRef;
+
+use api::lexer::{ Lexer, LexingResult };
+use api::error::{ SyntaxErrorCode, UnexpectedToken };
+
+/// TokenRef wrapper for `Lexer` that additionaly has position.
+#[derive(Debug)]
+pub struct ItemRef<'t> {
+    pub token: TokenRef<'t>,
+    pub position: Position,
+}
+
+/// Lexer token iterator.
+///
+/// 'i is iteration lifetime, or "one use of lexer".
+/// 't is template lifetime. It will live longer than this iteration.
+pub struct Tokens<'i, 't> {
+    /// Position of the next token to get.
+    next_pos: Position, // temporary field until I get cursor in.
+    env: &'i Lexer,
+    code: &'t str,
+}
+
+impl<'i, 't> Iterator for Tokens<'i, 't> {
+    type Item = LexingResult<ItemRef<'t>>;
+
+    fn next(&mut self) -> Option<LexingResult<ItemRef<'t>>> {
```

Comment:
OK, but we can definitely do it in a way that does not block future improvements.